arithmetic_analysis/newton_method.py

```diff
@@ -1,23 +1,50 @@
 """Newton's Method."""
 
 # Newton's Method - https://en.wikipedia.org/wiki/Newton%27s_method
-
-
-# function is the f(x) and function1 is the f'(x)
-def newton(function, function1, startingInt):
-    x_n = startingInt
+from typing import Callable
+
+
+# function is the f(x) and derivative is the f'(x)
+def newton(
+    function: Callable[[float], float],
```

Comment:
Is there a way to defined a type for `Callable[[float], float]` and then use that type to simplify the call parameters?

Reply:
Hmm... I think there is. I could define a Type Variable as described here: https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators
Let me try and see if it passes mypy.

Reply:
What do you know? It does work!